nncf/api/compression.py

```diff
@@ -315,6 +315,13 @@ def disable_scheduler(self) -> None:
         it to a dummy one that does not change the compression rate.
         """
 
+    @property
+    @abstractmethod
+    def maximal_compression_rate(self) -> float:
```

Comment:
Since accuracy aware algorithm uses common compression interface, we can't introduce it for specific algorithms and not introduce for others.
`maximal_compression_rate`=1 is valid for sparsity, it means there's no technical limitation to compress everything.

The only concern is that nobody needs maximal compression rate equal to 1. This model will not make any sense.
Probably we should call it `upper_bound_for_compression_rate` or something like that.

@nikita-savelyevv @alexsu52 What do you think about another approach when there's no `maximal_compression_rate`, but everything is handled by set_compression_rate?
We can return a specific error or exception that means we set more than the algorithm can really compress.

Reply:
@daniil-lyakhov @alexsu52, should `maximal_compression_rate` be different when some layers are in ignored scope for filter pruning by channels and for sparsity?

Reply:
In an offline discussion, it was decided to keep the introduction of `maximal_compression_rate` property as it may be useful in the future. The property should also be implemented for the magnitude sparsity compression algorithm, because for it, the property is not equal to 1.0 by default.